src/R/Editor/Impl/Completions/Providers/PackageFunctionCompletionProvider.cs

```diff
@@ -45,9 +45,10 @@ public class PackageFunctionCompletionProvider : IRCompletionListProvider, IRHel
         public IReadOnlyCollection<ICompletionEntry> GetEntries(IRIntellisenseContext context) {
             var completions = new List<ICompletionEntry>();
             var infoSource = _snippetInformationSource?.InformationSource;
-            var packages = GetPackages(context);
+            var packages = GetPackages(context).ToList();
```

Comment:
Can this method be called on UI thread?

Reply:
It is actually only called on UI thread

Reply:
Then you may want to use `ITaskService.Wait` for the `GetAllFilePackagesAsync`. That will prevent UI from freezing

Reply:
It returns `false` if we stopped waiting by timeout.